mlflow/openai/_openai_autolog.py

```diff
@@ -92,10 +92,26 @@ def _get_span_type(task) -> str:
     return span_type_mapping.get(task, SpanType.UNKNOWN)
 
 
+def _try_parse_raw_response(response: Any) -> Any:
+    """
+    As documented at https://github.com/openai/openai-python/tree/52357cff50bee57ef442e94d78a0de38b4173fc2?tab=readme-ov-file#accessing-raw-response-data-eg-headers,
+    a `LegacyAPIResponse` (https://github.com/openai/openai-python/blob/52357cff50bee57ef442e94d78a0de38b4173fc2/src/openai/_legacy_response.py#L45)
+    object is returned when the `create` method is invoked with `with_raw_response`.
+    `LegacyAPIResponse` is not exposed as a public class.
+    """
+    try:
+        return response.parse()
+    except Exception as e:
+        _logger.debug(f"Failed to parse {response} (type: {response.__class__}): {e}")
+
+    return response  # should be either a `pydantic.BaseModel` or a `openai.Stream` object
```

Comment:
Since we have `if not isinstance(raw_result, (BaseModel, Stream)):`, we should only reach here for unexpected inputs, not `BaseModel` or `Stream`?